### PR TITLE
Enable CI and fix failed test

### PR DESCRIPTION
**Pull request by ishikawa**
I noticed that the current `master` branch could not be passed by `mix test`. So I enabled a Github action like the parent [ex_aws](https://github.com/ex-aws/ex_aws/blob/master/.github/workflows/on-push.yml) repository.

### CODE REVIEW

File: test/lib/s3_test.exs [hunk 1]
```diff
@@ -29,7 +29,7 @@ defmodule ExAws.S3Test do
         "bucket",
         headers: %{"x-amz-request-payer" => "requester"},
         prefix: "/path/to/objs",
-        start_after: "/path/to/objs/sub",
+        start_after: "/path/to/objs/sub"
```
Comment (ishikawa) [inline test/lib/s3_test.exs:32]: This PR includes `mix format` diff a06c3c133edfaf6f37a3cb503bee99c573d58a92. I'll comment about where I changed intended.

File: test/lib/s3_test.exs [hunk 2]
```diff
@@ -369,8 +376,17 @@ defmodule ExAws.S3Test do
...
+
+    {:ok, url} =
+      S3.presigned_url(config(), :get, "bucket", "/foo/bar.txt", query_params: query_params)
+
+    assert_pre_signed_url(
+      url,
+      "https://s3.amazonaws.com/bucket/foo/bar.txt",
+      "3600",
+      query_params
+    )
```
Comment (ishikawa) [inline test/lib/s3_test.exs:389]: `S3.presigned_url/4` takes `:query_params` option.

File: test/lib/s3_test.exs [hunk 3]
```diff
@@ -449,21 +465,40 @@ defmodule ExAws.S3Test do
...
+        ) :: none()
+        when url: binary,
+             expected_scheme_host_path: binary,
+             expected_expire: pos_integer(),
+             expected_query_params: Access.t()
+  defp assert_pre_signed_url(
+         url,
+         expected_scheme_host_path,
+         expected_expire,
+         expected_query_params \\ []
```
Comment (ishikawa) [inline test/lib/s3_test.exs:482]: To the failed test work, I added a new options for testing query parameters.

**Comment by ishikawa** [comment]
The source branch has passed CI successfully. https://github.com/ishikawa/ex_aws_s3/runs/1686240557

**Comment by bernardd** [comment]
Fantastic, thanks @ishikawa - adding a workflow here was on my "todo list" but I hadn't had a chance to. Thanks for taking care of it.

**Pull request merged**